src/impl/peerconnection.cpp

```diff
@@ -439,24 +439,36 @@ void PeerConnection::rollbackLocalDescription() {
 	}
 }
 
-bool PeerConnection::checkFingerprint(const std::string &fingerprint) const {
+bool PeerConnection::checkFingerprint(const std::string &fingerprint, const CertificateFingerprint::Algorithm &algorithm) {
 	std::lock_guard lock(mRemoteDescriptionMutex);
 	if (!mRemoteDescription || !mRemoteDescription->fingerprint())
 		return false;
 
-	if (config.disableFingerprintVerification)
-		return true;
-
 	auto expectedFingerprint = mRemoteDescription->fingerprint()->value;
-	if (expectedFingerprint  == fingerprint) {
+	if (config.disableFingerprintVerification || expectedFingerprint == fingerprint) {
 		PLOG_VERBOSE << "Valid fingerprint \"" << fingerprint << "\"";
+		storeRemoteFingerprint(fingerprint, algorithm);
 		return true;
 	}
 
 	PLOG_ERROR << "Invalid fingerprint \"" << fingerprint << "\", expected \"" << expectedFingerprint << "\"";
 	return false;
 }
 
+void PeerConnection::storeRemoteFingerprint(const std::string &value, const CertificateFingerprint::Algorithm &algorithm) {
+	auto iter = std::find_if(rFingerprints.begin(), rFingerprints.end(), [&](const RemoteFingerprint& existing){return existing.value == value;});
+  bool seenPreviously = iter != rFingerprints.end();
+
+	if (seenPreviously) {
+		return;
+	}
+
+	rFingerprints.push_back({
+		value,
+		algorithm
+	});
+}
```

Comment:
This is not needed as the verifier callback will be called for a single algorithm.

Reply:
That's interesting, I was getting duplicate fingerprints as the callback was being invoked multiple times.

Reply:
It's called only once in the certificate callback of the cryptographic library, and always uses the fingerprint algorithm that was passed on DTLS transport initialization.